Three issues merge

rupole1185 opened the pull request:
This pull requests handles three different tasks:

1. "Slash" in strings within tests
2. Export of FDS file into cloud HPC
3. Mesh split

If you want we can go through it in order to check it together

emanuelegissi commented:
I reviewed the code you sent. As already said, this pull request touches several different features, and it is difficult for me to evaluate each of them separately. I prefer to receive atomic pull requests, one per feature. It is easier.

Anyway, I appreciate the overall work and wish to merge it, especially the link to CloudHPC and the split mesh op.

Here are my comments:

**CloudHPC**

For simplicity I don't like installing on the fly external pip packages:
`os.system("pip install requests")`
I prefer to use exclusively the packages that are already available in the Blender distribution.

The execute method does not upload the generated needed `.bingeom` and `.ge1` files. It is probably easier and more dependable to save in the `scene.bf_config_directory` or other tmp dir using the `export_scene.fds` operator, and then uploading all the resulting files to the cloud.

**Split mesh**

This feature is definitely useful!
I like a lot the idea, and the way you add modifiers to obtain that.

I want to think more about the way you integrated it. I would not use a `BFParam`, because there is no link with an FDS parameter. I would probably prefer an operator to act on an `ON_MESH` object.

Let's think a little more on how to best implement it.

**Verification suite**

I was quite unhappy with the flexibility of the first iteration of the Verification suite. I learned a lot from your contribution, and I rewrote it with an approach I know better (I am not a great dev, sorry).

Launch the `verification_new/verify.py` script and see by yourself.

It is a work in progress, but it already works for me, and allowed me to test the full V&V FDS suite importing.

**Import regex**

These days I refactored the import routines, thanks to the new verification suite. The brand new regexes should solve several issues.

Only 50 of the 2000 FDS V&V cases have import problems now.
And, in the meantime, I fixed a couple hundreds of cases that had typos in them in the official fds repo. 

I let the xml exporting behind for now, because I wanted something easily useable in a short time.

**linting (not very important ;-)**

The code you contribute should be linted with `pylint` and formatted with `black`.
Don't worry too much, I can do that for you.

Thank you very much, and I appreciate a lot your patience with me...
Emanuele

rupole1185 commented:
Ciao Emanuele,

so for the moment I took in charge the updates. In particular I am:

*CloudHPC*
Avoiding the use of pip. Regarding the two files  *.bingeom *and *.ge1* I
can upload them "together" with the .fds file. I can't at the moment upload
a whole folder just in once. To do that I should upload a compressed file,
which I something I would avoid. For this I will upload the three files
separately. As the files could become more and more we then could think of
a separate solution.

*Split Mesh*
I am moving to avoiding splitting the mesh object on the working tree. So
the tool will just work on the mesh object and "replace" it while writing
the FDS file with a "multiple" line of meshes.
After that, I'll start working on the MPI_PROCESS parameter. I just want to
stress that if we proceed in this was, MPI_PROCESS must be assigned by an
algorithm as the user can't manually assign it on the GUI since he won't
see the meshes. So this part will be mandatory.

Ruggero


On Mon, 31 Aug 2020 at 14:13, Emanuele Gissi <notifications@github.com>
wrote:

> I reviewed the code you sent. As already said, this pull request touches
> several different features, and it is difficult for me to evaluate each of
> them separately. I prefer to receive atomic pull requests, one per feature.
> It is easier.
>
> Anyway, I appreciate the overall work and wish to merge it, especially the
> link to CloudHPC and the split mesh op.
>
> Here are my comments:
>
> *CloudHPC*
>
> For simplicity I don't like installing on the fly external pip packages:
> os.system("pip install requests")
> I prefer to use exclusively the packages that are already available in the
> Blender distribution.
>
> The execute method does not upload the generated needed .bingeom and .ge1
> files. It is probably easier and more dependable to save in the
> scene.bf_config_directory or other tmp dir using the export_scene.fds
> operator, and then uploading all the resulting files to the cloud.
>
> *Split mesh*
>
> This feature is definitely useful!
> I like a lot the idea, and the way you add modifiers to obtain that.
>
> I want to think more about the way you integrated it. I would not use a
> BFParam, because there is no link with an FDS parameter. I would probably
> prefer an operator to act on an ON_MESH object.
>
> Let's think a little more on how to best implement it.
>
> *Verification suite*
>
> I was quite unhappy with the flexibility of the first iteration of the
> Verification suite. I learned a lot from your contribution, and I rewrote
> it with an approach I know better (I am not a great dev, sorry).
>
> Launch the verification_new/verify.py script and see by yourself.
>
> It is a work in progress, but it already works for me, and allowed me to
> test the full V&V FDS suite importing.
>
> *Import regex*
>
> These days I refactored the import routines, thanks to the new
> verification suite. The brand new regexes should solve several issues.
>
> Only 50 of the 2000 FDS V&V cases have import problems now.
> And, in the meantime, I fixed a couple hundreds of cases that had typos in
> them in the official fds repo.
>
> I let the xml exporting behind for now, because I wanted something easily
> useable in a short time.
>
> *linting (not very important ;-)*
>
> The code you contribute should be linted with pylint and formatted with
> black.
> Don't worry too much, I can do that for you.
>
> Thank you very much, and I appreciate a lot your patience with me...
> Emanuele
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firetools/blenderfds28x/pull/26#issuecomment-683740717>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXMUCKRKZGOCJKDSNI345TSDOHUXANCNFSM4QFZD3HQ>
> .
>

emanuelegissi commented:
Ok, thank you.

On Tue, Sep 15, 2020 at 10:22 AM Ruggero Poletto <notifications@github.com>
wrote:

> Ciao Emanuele,
>
> so for the moment I took in charge the updates. In particular I am:
>
> *CloudHPC*
> Avoiding the use of pip. Regarding the two files *.bingeom *and *.ge1* I
> can upload them "together" with the .fds file. I can't at the moment upload
> a whole folder just in once. To do that I should upload a compressed file,
> which I something I would avoid. For this I will upload the three files
> separately. As the files could become more and more we then could think of
> a separate solution.
>
> *Split Mesh*
> I am moving to avoiding splitting the mesh object on the working tree. So
> the tool will just work on the mesh object and "replace" it while writing
> the FDS file with a "multiple" line of meshes.
> After that, I'll start working on the MPI_PROCESS parameter. I just want to
> stress that if we proceed in this was, MPI_PROCESS must be assigned by an
> algorithm as the user can't manually assign it on the GUI since he won't
> see the meshes. So this part will be mandatory.
>
> Ruggero
>
>
> On Mon, 31 Aug 2020 at 14:13, Emanuele Gissi <notifications@github.com>
> wrote:
>
> > I reviewed the code you sent. As already said, this pull request touches
> > several different features, and it is difficult for me to evaluate each
> of
> > them separately. I prefer to receive atomic pull requests, one per
> feature.
> > It is easier.
> >
> > Anyway, I appreciate the overall work and wish to merge it, especially
> the
> > link to CloudHPC and the split mesh op.
> >
> > Here are my comments:
> >
> > *CloudHPC*
> >
> > For simplicity I don't like installing on the fly external pip packages:
> > os.system("pip install requests")
> > I prefer to use exclusively the packages that are already available in
> the
> > Blender distribution.
> >
> > The execute method does not upload the generated needed .bingeom and .ge1
> > files. It is probably easier and more dependable to save in the
> > scene.bf_config_directory or other tmp dir using the export_scene.fds
> > operator, and then uploading all the resulting files to the cloud.
> >
> > *Split mesh*
> >
> > This feature is definitely useful!
> > I like a lot the idea, and the way you add modifiers to obtain that.
> >
> > I want to think more about the way you integrated it. I would not use a
> > BFParam, because there is no link with an FDS parameter. I would probably
> > prefer an operator to act on an ON_MESH object.
> >
> > Let's think a little more on how to best implement it.
> >
> > *Verification suite*
> >
> > I was quite unhappy with the flexibility of the first iteration of the
> > Verification suite. I learned a lot from your contribution, and I rewrote
> > it with an approach I know better (I am not a great dev, sorry).
> >
> > Launch the verification_new/verify.py script and see by yourself.
> >
> > It is a work in progress, but it already works for me, and allowed me to
> > test the full V&V FDS suite importing.
> >
> > *Import regex*
> >
> > These days I refactored the import routines, thanks to the new
> > verification suite. The brand new regexes should solve several issues.
> >
> > Only 50 of the 2000 FDS V&V cases have import problems now.
> > And, in the meantime, I fixed a couple hundreds of cases that had typos
> in
> > them in the official fds repo.
> >
> > I let the xml exporting behind for now, because I wanted something easily
> > useable in a short time.
> >
> > *linting (not very important ;-)*
> >
> > The code you contribute should be linted with pylint and formatted with
> > black.
> > Don't worry too much, I can do that for you.
> >
> > Thank you very much, and I appreciate a lot your patience with me...
> > Emanuele
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/firetools/blenderfds28x/pull/26#issuecomment-683740717
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ABXMUCKRKZGOCJKDSNI345TSDOHUXANCNFSM4QFZD3HQ
> >
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/firetools/blenderfds28x/pull/26#issuecomment-692552510>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXMVY664QXKHO4LSPKI6LLSF4P5ZANCNFSM4QFZD3HQ>
> .
>
